src/xrpld/app/misc/detail/LendingHelpers.cpp

```diff
@@ -467,8 +489,6 @@ tryOverpayment(
     auto const newRounded = constructRoundedLoanState(
         totalValueOutstanding, principalOutstanding, managementFeeOutstanding);
 
-    newLoanProperties.totalValueOutstanding = newRounded.valueOutstanding;
```

Comment:
I added this in the most recent commit to update the `newLoanProperties` to accurate values because it is a parameter to `checkLoanGuards`, which needs accurate inputs.

```suggestion
// Update newLoanProperties so that checkLoanGuards can make an accurate evaluation.
newLoanProperties.totalValueOutstanding = newRounded.valueOutstanding;
```

Reply:
fixed!